packages/components/src/badge/stories/index.story.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryObj } from '@storybook/react';
+
+/**
+ * Internal dependencies
+ */
+import Badge from '..';
+
+const meta = {
+	component: Badge,
+	title: 'Components/Containers/Badge',
+	argTypes: {
+		className: {
+			control: { type: 'text' },
+		},
+		as: {
+			control: { type: 'select' },
+			options: [ 'div', 'span' ],
+		},
+		context: {
+			control: { type: 'select' },
+			options: [ 'neutral', 'info', 'warning', 'error', 'success' ],
+		},
+		children: {
+			control: { type: null },
+		},
```

Comment:
These can be handled by Storybook automatically, based on the type data. We want to avoid hardcoding these as much as possible, as it adds a maintenance burden.

```suggestion
	argTypes: {
		as: {
			control: { type: 'select' },
			options: [ 'div', 'span' ],
		},
```